### PR TITLE
[FrameworkBundle][Workflow] Fix deprectation when checking workflow.registry service in dump command

### PR DESCRIPTION
**Pull request by Jean-Beru**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no 
| BC breaks?    | no
| Deprecations? | no 
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        | 

This PR will declare `workflow.registry` as a public service to avoid deprecation when `WorkflowDumpCommand` checks its existence. It only concerns 3.4 since this `isEnabled` method will be removed in 4.0.

### CODE REVIEW

**Comment by stof** [comment]
Well, forcing to keep the service as public just for the BC layer is bad, as it means that any people using the service in a public way would not be warned about that (and so we cannot make the service private at all in 4.0).

A better solution would be to improve the way the command gets disabled in 3.4.

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/workflow.xml [hunk 1]
```diff
@@ -27,5 +27,9 @@
         <service id="Symfony\Component\Workflow\Registry" alias="workflow.registry" />
 
         <service id="workflow.security.expression_language" class="Symfony\Component\Workflow\EventListener\ExpressionLanguage" />
+
+        <service id="Symfony\Bundle\FrameworkBundle\Command\WorkflowDumpCommand">
```
Comment (ro0NL) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/workflow.xml]: moving this requires `$container->removeDefinition(WorkflowDumpCommand::class);` call to be updated, as console.xml is also conditionally loaded.

Comment (Jean-Beru) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/workflow.xml]: Nice catch ;-)

**Comment by Jean-Beru** [comment]
Thanks for your feedback @stof. So, I moved command in the conditionnally loaded `workflow.xml`.

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/workflow.xml (hunk 1 above)
Comment (ro0NL) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/workflow.xml]: well it still needs to be removed if console is not available ;-) not sure about the current effect of this service without console available. I guess it doesnt hurt.

Comment (Jean-Beru) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/workflow.xml]: Argh! I think we can keep command definition in `console.xml` and remove it if workflow is disabled.

**Review by ro0NL — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 2]
```diff
@@ -624,7 +624,7 @@ private function registerProfilerConfiguration(array $config, ContainerBuilder $
     private function registerWorkflowConfiguration(array $config, ContainerBuilder $container, XmlFileLoader $loader)
     {
         if (!$config['enabled']) {
-            if (!class_exists(Workflow\Workflow::class)) {
+            if ($container->hasDefinition(WorkflowDumpCommand::class)) {
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: the "if" can be removed: removeDefinition already does it internally

**Review by nicolas-grekas — APPROVED** [review]
(with minor comment)

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php (hunk 2 above)
Comment (Jean-Beru) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: Correct. Thks @nicolas-grekas !

**Review by nicolas-grekas — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @Jean-Beru.

**Pull request closed without merging**